main/src/ca/uwaterloo/flix/language/phase/unification/BoolFormula.scala

```diff
@@ -230,6 +230,56 @@ object BoolFormula {
 
     override def isFalse(f: BoolFormula): Boolean = f == BoolFormula.False
 
+    override def isVar(f: BoolFormula): Boolean = f match {
+      case Var(_) => true
+      case _ => false
+    }
+
+    override def satisfiable(f: BoolFormula): Boolean = f match {
+      case BoolFormula.True => true
+      case BoolFormula.False => false
+      case _ => {
+        val vars = freeVars(f)
+        val assignments = enumerateAssignments(vars)
```

Comment:
We prefer functional-style. Presumably this can just be `enumerate(...).exists(checkAssignment)` or something like that.